web/src/lottie-player.model.ts

```diff
@@ -0,0 +1,199 @@
+import { LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+
+// Define library version
+export interface LibraryVersion {
+  THORVG_VERSION: string
+}
+
+// Define file type which can be exported
+export enum ExportableType {
+  GIF = 'gif',
+}
+
+// Define valid player states
+export enum PlayerState {
+  Destroyed = "destroyed", // Player is destroyed by `destroy()` method
+  Error = "error", // An error occurred
+  Loading = "loading", // Player is loading
+  Paused = "paused", // Player is paused
+  Playing = "playing", // Player is playing
+  Stopped = "stopped",  // Player is stopped
+}
+
+// Define play modes
+export enum PlayMode {
+  Bounce = "bounce",
+  Normal = "normal",
+}
+
+// Define player events
+export enum PlayerEvent {
+  Complete = "complete",
+  Destroyed = "destroyed",
+  Error = "error",
+  Frame = "frame",
+  Load = "load",
+  Loop = "loop",
+  Pause = "pause",
+  Play = "play",
+  Ready = "ready",
+  Stop = "stop",
+}
+
+export abstract class LottiePlayerModel extends LitElement {
```

Comment:
What is a practical benefit of this abstract model?

Reply:
Basically unnecessary and it's preference thing, I separated file because I'm worried about the file's going to be too long and hard to read. I thought separated interface (abstract class) is simpler to be read.

Technically, there's no benefit except code quiality.